internal/app/machined/pkg/system/runner/containerd/containerd.go

```diff
@@ -137,6 +137,24 @@ func (c *containerdRunner) Run(eventSink events.Recorder) error {
 	// attempt to clean up a task if it already exists
 	task, err = c.container.Task(c.ctx, nil)
 	if err == nil {
+		var s <-chan containerd.ExitStatus
+
+		s, err = task.Wait(c.ctx)
```

Comment:
do we need some timeout, like in a case of reboot/reset and the task hangs forever, or do we have a top level context that cancels this?

Reply:
it should be the `<-c.stop` receive below which will be triggered if something tries to stop the service

`Wait` itself is asynchronous - it returns immediately